segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/ServerConnectionInboundHandler.java

```diff
@@ -66,15 +66,11 @@ public void send(WireCommand cmd) {
         Channel c = getChannel();
         // Work around for https://github.com/netty/netty/issues/3246
         EventLoop eventLoop = c.eventLoop();
-        if (eventLoop.inEventLoop()) {
-            eventLoop.execute(() -> writeAndFlush(c, cmd));
-        } else {
-            writeAndFlush(c, cmd);
-        }
+        eventLoop.execute(() -> write(c, cmd));
     }
 
-    private static void writeAndFlush(Channel channel, WireCommand data) {
-        channel.writeAndFlush(data).addListener(ChannelFutureListener.FIRE_EXCEPTION_ON_FAILURE);
+    private static void write(Channel channel, WireCommand data) {
+        channel.write(data).addListener(ChannelFutureListener.CLOSE_ON_FAILURE);
```

Comment:
In case of an exception `io.pravega.segmentstore.server.host.handler.ServerConnectionInboundHandler#exceptionCaught` would be invoked, this would also close the connection.